designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala

```diff
@@ -109,10 +116,69 @@ class DeploymentServiceSpec
       processValidatorByProcessingType,
       TestFactory.scenarioResolverByProcessingType,
       listener,
-      scenarioStateTimeout = scenarioStateTimeout
+      scenarioStateTimeout = scenarioStateTimeout,
+      deploymentCommentSettings
     )
   }
 
+  // TODO: temporary step - we would like to extract the validation and the comment validation tests to external validators
+
+  private val nonEmptyCommentSettings = Some(createNonEmptyDeploymentCommentSettings())
+
+  def createDeploymentServiceWithCommentSettings(): DeploymentService = {
+    createDeploymentService(deploymentCommentSettings = nonEmptyCommentSettings)
+  }
+
+  test { "should return error when trying to deploy without comment when comment is required" } {
+    val deploymentServiceWithCommentSettings = createDeploymentServiceWithCommentSettings()
+    val processName: ProcessName             = generateProcessName
+    val id                                   = prepareProcess(processName).dbioActionValues
+
+    val result = deploymentServiceWithCommentSettings.deployProcessAsync(id, None, None).failed.futureValue
+
+    result shouldBe a[ValidationError]
+    result.getMessage.trim shouldBe "Comment is required."
+  }
+
+  test { "should pass when having an ok comment" } {
+    val deploymentServiceWithCommentSettings = createDeploymentServiceWithCommentSettings()
+    val processName: ProcessName             = generateProcessName
+    val id                                   = prepareProcess(processName).dbioActionValues
+
+    deploymentManager.withWaitForDeployFinish(processName) {
+      deploymentServiceWithCommentSettings.deployProcessAsync(id, None, Some("samplePattern")).futureValue
+      deploymentServiceWithCommentSettings
```

Comment:
Do not need to check `SimpleStateStatus.DuringDeploy`, we need to assert that we eventually landed in `SimpleStateStatus.Running`.